src/main/java/cuchaz/enigma/analysis/index/ReferenceIndex.java

```diff
@@ -97,7 +97,8 @@ public void processIndex(JarIndex index) {
 	}
 
 	private <K extends Entry<?>, V extends Entry<?>> Multimap<K, V> remapReferences(JarIndex index, Multimap<K, V> multimap) {
-		Multimap<K, V> resolved = HashMultimap.create(multimap.keySet().size(), multimap.size() / multimap.keySet().size());
+		final int keySetSize = multimap.keySet().size();
```

Comment:
Should just return `ImmutableMultimap.of()` when the input map is empty.

Reply:
There's a call to `put` on the return value in this class. I'm not sure if it's ever called if the map is empty anyway, but why risk it? Also, this is exactly how the other method works, so presumably there's some reason not to.

Reply:
Then the line below should be 
```java
Multimap<K, V> resolved = HashMultimap.create(keySetSize, keySetSize == 0 ? 0 : multimap.size() / keySetSize);
```
to reduce a keyset size call.

Reply:
Uhh my bad, when the input map is empty, just return the input map lol (it's going to be an empty hash multi map)

Reply:
Again, still not the safest idea, since now mutating one mutates the other. I really don't see why you have a problem with the code as it is.

Reply:
Again, I am not sure why you insist creating a new map and return when you can just return the old map (because there is literally nothing to remap in the old map, and there is no reason to eat memory and make a new map!)